### PR TITLE
Review documentation on logging

### PR DESCRIPTION
**Pull request by asteflova**
TL;DR: The changes introduced by this PR revolve most importantly around consistency and eliminating duplication, in some cases also around applying more appropriate markup. Please review and comment so that we can make this content as useful to users as possible.

For a more detailed overview of why I propose the changes, read on:

Consistency:
* I introduce commands in the same way and generally display the same type of information in the same places.
* This is important because the chapter provides mostly referential information on logging (lists of commands to use), so consistent structure and parallelism help users scan the content and quickly identify the bits they need.

Eliminating duplication:
* I identified and removed some cases of duplication of the actual content (e.g. the same command documented in two places for the same purpose).
* Where possible, I now link to in-project sources like config files for lists of options, man pages for command documentation, etc. Lists of files, command options, and such are important, but because it is a huge overhead to keep them up-to-date, linking to their primary source is the preferred option. (Trust the Source, Luke!)

Mark up
* I replaced some tables with lists (because [Let's bring table to the table again](https://heyoka.medium.com/lets-bring-table-to-the-table-again-f1ae751159d5 )), some bullet points with procedure steps, added missing markup here and there.

* [X] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [X] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (planned Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* [ ] Foreman 3.2/Katello 4.4 on EL7 & EL8
* [ ] Foreman 3.1/Katello 4.3 on EL7 & EL8 (Satellite 6.11 EL7/8; orcharhino 6.3 on EL7/8)
* We do not accept PRs for Foreman older than 3.1.

### CODE REVIEW

File: guides/common/modules/ref_overview-of-individual-loggers.adoc [hunk 1]
```diff
@@ -1,53 +1,54 @@
-[id="Enabling_Individual_Loggers_{context}"]
-= Enabling Individual Loggers
-
-You can enable individual loggers for selective logging.
-{Project} uses the following loggers:
+[id="Overview_of_Individual_Loggers_{context}"]
+= Overview of Individual Loggers
+The following list provides an overview of selected loggers. You can find the complete list of loggers with their default values in
```
Comment (ekohl) [inline guides/common/modules/ref_overview-of-individual-loggers.adoc:3]: Some thought: I think it's a poor practice to suggest users to read the source code. Perhaps we can create a `foreman-rake logging:list` (or similarly named) task. Or at least some supported way. Perhaps even generate the whole table.

File: guides/common/assembly_logging-and-reporting-problems.adoc [hunk 2]
```diff
@@ -1,19 +1,23 @@
 include::modules/con_logging-and-reporting-problems.adoc[]
 
-include::modules/proc_enabling-debug-logging.adoc[leveloffset=+1]
-
 include::modules/proc_increasing-the-logging-levels-to-help-with-debugging.adoc[leveloffset=+1]
 
+ifndef::foreman-deb[]
+include::modules/ref_configuring-logging-levels-with-hammer-cli.adoc[leveloffset=+1]
```
Comment (ekohl) [inline guides/common/assembly_logging-and-reporting-problems.adoc:6]: For context: hammer-cli-admin is a really weird package. Normally Hammer CLI is used to talk to the Foreman REST API, so it can be used remotely. hammer-cli-admin is the very odd exception in that it is completely different: it performs actions on the local system.

On Satellite it is installed by default, but in upstream it isn't. There is no installer support either, so you have to know the package name.

So technically you should have instructions to install it, but I'd really like to discourage people from using it.

Perhaps nothing actionable right now, but providing some context.

File: guides/common/modules/ref_overview-of-individual-loggers.adoc (hunk 1 above)
Comment (asteflova) [reply guides/common/modules/ref_overview-of-individual-loggers.adoc:3]: Regarding referencing source code: I get what you mean, it's not a pretty solution. But when I weighed the pros and cons, I couldn't really come up with a logical reason not to do it. So I decided to keep the reference.

In general, Red Hat writers are constantly being asked to reduce the amount of content we maintain. This is one way to do that -- to avoid listing options, components, etc. and instead link to their primary source wherever possible.

For the record, I filed https://github.com/theforeman/foreman-documentation/issues/2403 about this.

**Review by ekohl — CHANGES_REQUESTED** [review]
Looking at https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_smartproxy I see it is still done manually. The proper way is very similar to https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_the_foreman_application and is done using `--foreman-proxy-log-level DEBUG` and reset using `--reset-foreman-proxy-log-level`.

https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_redis is a bigger case: this section only applies to EL7, which is no longer supported. The service is now called `redis` instead of `rh-redis5-redis`. I don't think we expose the redis log level today, but the installer does manage it so it will be reverted.

https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_postgres is a poor section. Just enabling all statements to be logged is pointless IMHO. You're not going to find anything useful, unless you really know what you want. I question if it makes sense to keep this at all.

https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_puppet_agent it surprises me we don't have any parameter, but I think you can use `--puppet-agent-additional-settings log_level:debug`.

https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_puppet_server very similar to the agent: `--puppet-server-additional-settings log_level:debug`.

File: guides/common/modules/ref_log-file-directories.adoc [hunk 3]
```diff
@@ -1,23 +1,19 @@
...
-| `/var/log/messages` | Various other log messages
-| `/var/log/puppetlabs/puppet` | Configuration management
-| `/var/log/rhsm` | Subscription management
-| `/var/log/tomcat` | Candlepin webservice logs
-|===
-
-You can also use the `foreman-tail` command to follow many of the log files related to {Project}.
-You can run `foreman-tail -l` to list the processes and services that it follows.
+| Log file directory | Description of log file content
+| `/var/log/candlepin/` | Subscription management
```
[the remark this answers is not in the document]
Comment (asteflova) [reply guides/common/modules/ref_log-file-directories.adoc:9]: Added an ifdef for katello,orcharhino,satellite.

File: guides/common/modules/ref_log-file-directories.adoc [hunk 4]
```diff
@@ -1,23 +1,19 @@
...
+| `/var/log/foreman-installer/` | Installer
+| `/var/log/foreman-maintain/` | Foreman maintain
+| `/var/log/foreman-proxy/` | Foreman proxy
+| `/var/log/foreman/` | Foreman
+| `/var/log/httpd/` | Apache HTTP server
+| `/var/log/messages/` | Various other log messages
+| `/var/log/puppetlabs/puppet/` | Configuration management
+| `/var/log/rhsm/` | Subscription management
```
[the remark this answers is not in the document]
Comment (asteflova) [reply guides/common/modules/ref_log-file-directories.adoc]: Deleted this line. The section is not designed to list all log directories, and considering the feedback, it makes sense to me not to include this one.

File: guides/common/modules/ref_log-file-directories.adoc [hunk 5]
```diff
@@ -1,23 +1,19 @@
...
+| `/var/log/tomcat/` | Candlepin webservice logs
```
[the remark this answers is not in the document]
Comment (asteflova) [reply guides/common/modules/ref_log-file-directories.adoc]: Added an ifdef for satellite,katello,orcharhino.

File: guides/common/modules/ref_log-file-directories.adoc [hunk 6]
```diff
@@ -1,23 +1,19 @@
...
-
-You can also use the `foreman-tail` command to follow many of the log files related to {Project}.
-You can run `foreman-tail -l` to list the processes and services that it follows.
+| Log file directory | Description of log file content
+| `/var/log/candlepin/` | Subscription management
+| `/var/log/foreman-installer/` | Installer
+| `/var/log/foreman-maintain/` | Foreman maintain
+| `/var/log/foreman-proxy/` | Foreman proxy
+| `/var/log/foreman/` | Foreman
+| `/var/log/httpd/` | Apache HTTP server
```
[the remark this answers is not in the document]
Comment (asteflova) [reply guides/common/modules/ref_log-file-directories.adoc]: Added the attribute.

**Comment by asteflova** [comment]
> Looking at https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_smartproxy I see it is still done manually. The proper way is very similar to https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_the_foreman_application and is done using `--foreman-proxy-log-level DEBUG` and reset using `--reset-foreman-proxy-log-level`.
> 
> https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_puppet_agent it surprises me we don't have any parameter, but I think you can use `--puppet-agent-additional-settings log_level:debug`.
> 
> https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_puppet_server very similar to the agent: `--puppet-server-additional-settings log_level:debug`.

Updated all three.

**Comment by asteflova** [comment]
> https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_postgres is a poor section. Just enabling all statements to be logged is pointless IMHO. You're not going to find anything useful, unless you really know what you want. I question if it makes sense to keep this at all.

Removed.

**Comment by asteflova** [comment]
> https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_redis is a bigger case: this section only applies to EL7, which is no longer supported. The service is now called `redis` instead of `rh-redis5-redis`. I don't think we expose the redis log level today, but the installer does manage it so it will be reverted.

If the Redis section applies only to unsupported scenarios, I think it can be safely deleted. Or have I mistunderstood @ekohl?

In addition, I should check the docs for other instances of `rh-redis5-redis`. I can see at least one in this section.

**Comment by ekohl** [comment]
> > https://theforeman-foreman-documentation-preview-pr-2402.surge.sh/nightly/Administering_Project/index-foreman-deb.html#_increasing_the_logging_level_for_redis is a bigger case: this section only applies to EL7, which is no longer supported. The service is now called `redis` instead of `rh-redis5-redis`. I don't think we expose the redis log level today, but the installer does manage it so it will be reverted.
> 
> If the Redis section applies only to unsupported scenarios, I think it can be safely deleted. Or have I mistunderstood @ekohl?
> 
> In addition, I should check the docs for other instances of `rh-redis5-redis`. I can see at least one in this section.

You misunderstood. On EL7 we used Software Collections (SCL) to get Redis 5. That meant we had a package name `rh-redis5-redis`. On EL8 the package name (and service name as well as log files) is plain `redis`. We've dropped EL7 support (see our PR template for the versions), so the SCL variant is no longer correct.

File: guides/common/assembly_logging-and-reporting-problems.adoc (hunk 2 above)
Comment (asteflova) [reply guides/common/assembly_logging-and-reporting-problems.adoc:6]: Nothing actionable here, resolving.

File: guides/common/modules/ref_overview-of-individual-loggers.adoc (hunk 1 above)
Comment (asteflova) [reply guides/common/modules/ref_overview-of-individual-loggers.adoc:3]: Nothing actionable here right now, as far as I can see (there is an agreement on a short-term solution in this thread). Resolving. Feel free to correct me if I got it wrong.

**Comment by asteflova** [comment]
@ekohl Are you okay with the current state? If you have any more feedback, I'll be happy to work on it.

File: guides/common/modules/ref_utilities-for-processing-log-information.adoc [hunk 7]
```diff
@@ -1,37 +1,49 @@
...
-
++
 [IMPORTANT]
 ====
-The collection process removes security information such as passwords, tokens, and keys while collecting information.
+ifdef::foreman-el,katello,satellite[]
+The `sosreport` command
+endif::[]
+ifdef::foreman-deb,orcharhino[]
+The `foreman-debug` command
```
Comment (ekohl) [inline guides/common/modules/ref_utilities-for-processing-log-information.adoc:39]: Looking at line 30, Shouldn't this be `orcharhino-debug` for Orcharhino?

cc @maximiliankolb

**Review by ekohl — APPROVED** [review]
One small note, but overall :+1:

File: guides/common/modules/ref_utilities-for-processing-log-information.adoc (hunk 7 above)
Comment (maximiliankolb) [reply guides/common/modules/ref_utilities-for-processing-log-information.adoc:39]: Thanks for the ping! Yes, the command is `orcharhino-debug`.

Comment (asteflova) [reply guides/common/modules/ref_utilities-for-processing-log-information.adoc:39]: Addressed.

File: guides/common/modules/ref_utilities-for-processing-log-information.adoc [hunk 8]
```diff
@@ -1,37 +1,52 @@
...
 ifdef::foreman-deb,orcharhino[]
 foreman-debug::
 The `foreman-debug` command collects configuration and log file data for {ProjectName}, its back-end services, and system information.
-This information is collected and written to a tar file.
-By default, the output tar file is located at `/tmp/__foreman-debug-xxx.tar.xz__`.
-
+This output is stored in a tar file.
+By default, the tar file is located at `/tmp/__foreman-debug-xxx.tar.xz__`.
++
 Additionally, the `foreman-debug` command exports tasks run during the last 60 days.
```
Comment (ekohl) [inline guides/common/modules/ref_utilities-for-processing-log-information.adoc:19]: Should this also be changed to `orcharhino-debug`? Would an attribute help here?

**Review by ekohl — COMMENTED** [review]
@maximiliankolb just a general note: I'd recommend ATIX to also switch over to sosreport. It can generate a tarball that's more complete. Users can then send that just like they send the `foreman-debug` one today. We've been talking about deprecating and removing `foreman-debug`, so coming up with a replacement plan would be nice.

The reason we don't use it on Debian today is that the version in Debian 11 is too old to include our specific patches.

File: guides/common/modules/ref_utilities-for-processing-log-information.adoc (hunk 8 above)
Comment (asteflova) [reply guides/common/modules/ref_utilities-for-processing-log-information.adoc:19]: Sorry, I should have noticed there was more. I'll take a closer look at the whole file, an attribute might indeed be a good solution.

Comment (asteflova) [reply guides/common/modules/ref_utilities-for-processing-log-information.adoc]: Addressed in the latest commit: a new attribute named :project-debug: that resolves to either foreman-debug or orcharhino-debug.

**Review by ekohl — APPROVED** [review]
I'll let @maximiliankolb do the final review on orcharhino-debug, but :+1: from my side.

**Comment by asteflova** [comment]
Tech review done, but I'm wondering about style review. @maximiliankolb can I assume your suggestions on this PR so far are sufficient to pass the style review check?

Technically, this whole PR is supposed to be a style review for old content, so I'm just wondering how to use the "style review" labels for this type of update.

File: guides/common/modules/proc_retrieving-status-of-services.adoc [hunk 9]
```diff
@@ -1,28 +1,35 @@
...
 
-.Procedure
-. In the {ProjectWebUI}, navigate to *Administer > About*.
-. On the *Smart Proxies* tab, you can view the status of all {SmartProxies}.
-. On the *Compute Resources* tab, you can view the status of attached compute resource providers.
-. In the *Backend System Status* table, you can view the status of all back-end services.
+{Project} uses a set of back-end services.
+When troubleshooting, you can check the status of {Project} services.
+
+.Web UI procedure
```
Comment (maximiliankolb) [inline guides/common/modules/proc_retrieving-status-of-services.adoc:7]:
```suggestion
.Procedure
```

Also, just FYI, there's a different name for the browser-based interface in downstream orcharhino: we call it "management UI". > That's why there's an attribute "ProjectWebUI".

File: guides/common/modules/ref_configuring-logging-levels-with-hammer-cli.adoc [hunk 10]
```diff
@@ -0,0 +1,51 @@
+[id="Configuring-Logging-Levels-with-Hammer-CLI_{context}"]
+= Configuring Logging Levels with Hammer CLI
+
+You can use Hammer to log debugging information for various Satellite components.
```
Comment (maximiliankolb) [inline guides/common/modules/ref_configuring-logging-levels-with-hammer-cli.adoc:4]:
```suggestion
You can use Hammer to log debugging information for various {Project} components.
```

File: guides/common/modules/ref_configuring-logging-levels-with-hammer-cli.adoc [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+----
+# hammer admin logging --all --level-production
+# {foreman-maintain} service restart
+----
+
+* To set `DEBUG` level logging for a specific component:
++
+[options="nowrap" subs="+quotes"]
+----
+# hammer admin logging --components _component_ --level-debug
```
Comment (maximiliankolb) [inline guides/common/modules/ref_configuring-logging-levels-with-hammer-cli.adoc:34]: no "foreman-maintain service restart" here?

File: guides/common/modules/ref_configuring-logging-levels-with-hammer-cli.adoc [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+----
+
+* To list all available logging options:
++
+[options="nowrap"]
+----
+# hammer admin logging --help
+
+Usage:
+    hammer admin logging [OPTIONS]
```
Comment (maximiliankolb) [inline guides/common/modules/ref_configuring-logging-levels-with-hammer-cli.adoc:44]:
```suggestion
```
We generally do not show stdout from commands that a user runs.

File: guides/common/modules/ref_log-file-directories.adoc [hunk 13]
```diff
@@ -1,23 +1,22 @@
 [id="Log_File_Directories_{context}"]
 = Log File Directories Provided by {Project}
```
Comment (maximiliankolb) [inline guides/common/modules/ref_log-file-directories.adoc:2]:
```suggestion
= Log File Directories
```

Alternative: "Log File Directories on {ProjectServer}". Your call; but please ensure file name, anchor, and title matches.

File: guides/common/modules/ref_log-file-directories.adoc [hunk 14]
```diff
@@ -1,23 +1,22 @@
...
 
-{ProjectName} provides system information in the form of notifications and log files.
+The following table provides an overview of selected log directories {ProjectName} provides.
```
Comment (maximiliankolb) [inline guides/common/modules/ref_log-file-directories.adoc:4]:
```suggestion
The following table provides an overview of selected log directories {Project} provides.
```

File: guides/common/modules/ref_overview-of-individual-loggers.adoc (hunk 1 above)
Comment (maximiliankolb) [inline guides/common/modules/ref_overview-of-individual-loggers.adoc:3]:
```suggestion
The following list provides an overview of selected loggers.
You can find the complete list of loggers with their default values in
```

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Made some minor comments and suggestions.

File: guides/common/modules/ref_log-file-directories.adoc (hunk 13 above)
Comment (asteflova) [reply guides/common/modules/ref_log-file-directories.adoc]: Applied.

File: guides/common/modules/ref_configuring-logging-levels-with-hammer-cli.adoc [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+----
+# hammer admin logging --all --level-production
+# {foreman-maintain} service restart
+----
+
+* To set `DEBUG` level logging for a specific component:
++
+[options="nowrap" subs="+quotes"]
+----
+# hammer admin logging --components _component_ --level-debug
```
Comment (maximiliankolb) [inline guides/common/modules/ref_configuring-logging-levels-with-hammer-cli.adoc:34]:
```suggestion
# hammer admin logging --components _My_Component_ --level-debug
```

File: guides/common/modules/ref_log-file-directories.adoc [hunk 16]
```diff
@@ -1,23 +1,22 @@
...
+ifdef::satellite,katello,orcharhino[]
+| `/var/log/candlepin/` | Subscription management
+endif::[]
+| `/var/log/foreman-installer/` | Installer
+| `/var/log/foreman-maintain/` | Foreman maintain
+| `/var/log/foreman-proxy/` | Foreman proxy
+| `/var/log/foreman/` | Foreman
+| `/var/log/{apache-log-dir}/` | Apache HTTP server
+| `/var/log/messages/` | Various other log messages
+| `/var/log/puppetlabs/puppet/` | Configuration management
```
Comment (maximiliankolb) [inline guides/common/modules/ref_log-file-directories.adoc:18]:
```suggestion
| `/var/log/puppetlabs/puppet/` | Puppet
```

File: guides/common/modules/ref_overview-of-individual-loggers.adoc [hunk 17]
```diff
@@ -1,53 +1,55 @@
-[id="Enabling_Individual_Loggers_{context}"]
-= Enabling Individual Loggers
-
-You can enable individual loggers for selective logging.
-{Project} uses the following loggers:
+[id="Overview_of_Individual_Loggers_{context}"]
+= Overview of Individual Loggers
+The following list provides an overview of selected loggers.
```
Comment (maximiliankolb) [inline guides/common/modules/ref_overview-of-individual-loggers.adoc:3]:
```suggestion

The following list provides an overview of selected loggers.
```

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
In the list of directories, would it make sense to apply branding for the second column? like ".../foreman-installer | {foreman-installer}". Your call, I am OK either way.

Rest LGTM!

**Comment by asteflova** [comment]
> In the list of directories, would it make sense to apply branding for the second column? like ".../foreman-installer | {foreman-installer}". Your call, I am OK either way.

I tried it, but didn't like the way it looked. So let's keep it as it is.
 
> Rest LGTM!

I guess it can be merged then, right?

**Review by maximiliankolb — APPROVED** [review]
Thanks Anet, LGTM!

**Comment by maximiliankolb** [comment]
@asteflova Can I squash everyone into a single commit?

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: :heavy_check_mark: 
   28979ce39..13419dbd7  3.8 -> 3.8
   bb72a570c..30c51e4c6  3.7 -> 3.7

**Comment by ekohl** [comment]
Thanks for pushing this over the finish line @asteflova. I submitted https://github.com/theforeman/foreman-documentation/pull/2471 as a follow up, using the `--katello-candlepin-loggers` installer option in nightly.

**Comment by asteflova** [comment]
> Thanks for pushing this over the finish line @asteflova.

No problem, it was actually fun to get to work on this :upside_down_face: Thank you for the reviews!
